File: Programming Basics with Python/Вложени цикли/sum_of_two_numbers.py

interval_start = int(input())
interval_end = int(input())
input_magic_num = int(input())

combination = 0
magic_num = False
for x1 in range(interval_start, interval_end+1):
    for x2 in range(interval_start, interval_end+1):
        combination += 1
        if x1 + x2 == input_magic_num:
            print(f"Combination N:{combination} ({x1} + {x2} = {input_magic_num})")
            magic_num = True

    if magic_num:
        break

if not magic_num:
    print(f"{combination} combinations - neither equals {input_magic_num}")

File: Programming Basics with Java/Nested Loops/SumOfTwoNumbers.java
import java.util.Scanner;

public class SumOfTwoNumbers {
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);

        int start = Integer.parseInt(input.nextLine());
        int end = Integer.parseInt(input.nextLine());
        int magicNumber = Integer.parseInt(input.nextLine());

        int count = 0;
        boolean found = false;

        for (int i = start; i <= end; i++) 
        {
            for (int j = start; j <= end; j++) 
            {
                count++;
                int sum = i + j;
                if (sum == magicNumber) 
                {
                    found = true;
                    System.out.printf("Combination N:%d (%d + %d = %d)%n", count, i, j, magicNumber);
                    break;
                }
            }

            if (found) 
            {
                break;
            }
        }

        if (!found) 
        {
            System.out.printf("%d combinations - neither equals %d%n", count, magicNumber);
        }
    }
}
